File: src/test/java/com/binance/api/examples/MarginAccountEndpointsExample.java
package com.binance.api.examples;

import com.binance.api.client.BinanceApiClientFactory;
import com.binance.api.client.BinanceApiMarginRestClient;
import com.binance.api.client.domain.account.MarginAccount;
import com.binance.api.client.domain.account.Trade;

import java.util.List;

/**
 * Examples on how to get margin account information.
 */
public class MarginAccountEndpointsExample {

  public static void main(String[] args) {
    BinanceApiClientFactory factory = BinanceApiClientFactory.newInstance("YOUR_API_KEY", "YOUR_SECRET");
    BinanceApiMarginRestClient client = factory.newMarginRestClient();

    // Get account balances
    MarginAccount marginAccount = client.getAccount();
    System.out.println(marginAccount.getUserAssets());
    System.out.println(marginAccount.getAssetBalance("ETH"));
    System.out.println(marginAccount.getMarginLevel());

    // Get list of trades
    List<Trade> myTrades = client.getMyTrades("NEOETH");
    System.out.println(myTrades);
  }
}

File: src/test/kotlin/com/binance/api/examples/MarginAccountEndpointsExample.kt
package com.binance.api.examples

import com.binance.api.client.BinanceApiClientFactory.Companion.newInstance
import com.binance.api.client.domain.account.MarginAccount

/**
 * Examples on how to get margin account information.
 */
class MarginAccountEndpointsExample {
    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val factory = newInstance("YOUR_API_KEY", "YOUR_SECRET")
            val client = factory.newMarginRestClient()

            // Get account balances
            val marginAccount: MarginAccount = client.getAccount()
            println(marginAccount.userAssets)
            println(marginAccount.getAssetBalance("ETH"))
            println(marginAccount.marginLevel)

            // Get list of trades
            val myTrades = client.getMyTrades("NEOETH")
            println(myTrades)
        }
    }
}